Corrected length of start signal for DHT22.

noisyfly opened the pull request:
When I was testing original source using AM2302 I had, from time to time, conversions errors. So I connected my scope to data-line. The start signal was very short I don't remember now but it was about 20-40us and my AM2302 doesn't start transmission in return. In code is comment that intention was to generate 2ms impulse. But the same variable is used to count 1s spacing between conversions so the condition is always fulfilled. Corrected code is nearly the same only slightly modified. 
Additionally I expanded spacing between successive conversions from 1 second to 2 seconds, base on AM2302 documentation, now it is included in #define. But 1s spacing was working fine for me.

Commit:
Corrected length of start signal for DHT22. Additionaly extended spacing between sucessive conversions from 1 second to 2 seconds, base on documentation, now it is included in #define.

danjperron commented:
Thanks! I will check what you did when I'll be back . I'm in Ireland for new year.
Happy new Year!

danjperron commented:
Ok I corrected the code for the DHT22.

I simply inserted 'Timerms=0' in DHT22CycleIdle().

I also added the parasite mode for the DS18B20 and increased the read position around 14us instead of 1us. 
This allow to read the sensor on long cable when the capacitance is high.

And I fixed all the warnings for the XC8 compiler.



Daniel Perron



Le 30/12/17 à 17:06, Grzegorz Chrzanowski  <notifications@github.com> a écrit :
> 
> 
> When I was testing original source using AM2302 I had, from time to time, conversions errors. So I connected my scope to data-line. The start signal was very short I don't remember now but it was about 20-40us and my AM2302 doesn't start transmission in return. In code is comment that intention was to generate 2ms impulse. But the same variable is used to count 1s spacing between conversions so the condition is always fulfilled. Corrected code is nearly the same only slightly modified.
> 
> Additionally I expanded spacing between successive conversions from 1 second to 2 seconds, base on AM2302 documentation, now it is included in #define. But 1s spacing was working fine for me.
>   You can view, comment on, or merge this pull request online at: 
>  https://github.com/danjperron/PicMultiPurpose/pull/3
>  Commit Summary 
>  	Corrected length of start signal for DHT22. Additionaly extended spacing between sucessive conversions from 1 second to 2 seconds, base on documentation, now it is included in #define.
>  
>  File Changes 
>  	 M DHT22.c(https://github.com/danjperron/PicMultiPurpose/pull/3/files#diff-0) (6) 
>  	 M DHT22.h(https://github.com/danjperron/PicMultiPurpose/pull/3/files#diff-1) (2) 
>  	 M main.c(https://github.com/danjperron/PicMultiPurpose/pull/3/files#diff-2) (2) 
>  
>  Patch Links: 
>  	https://github.com/danjperron/PicMultiPurpose/pull/3.patch
>  	https://github.com/danjperron/PicMultiPurpose/pull/3.diff
>  
>  
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub(https://github.com/danjperron/PicMultiPurpose/pull/3), or mute the thread(https://github.com/notifications/unsubscribe-auth/AEJf3MUIOkkykLPIt1sux50ApwlbU8Gqks5tFrPOgaJpZM4RPtS9).
> 
> 
> 
> 
>   
> 
>  
>  
>

noisyfly commented:
Inserting 'Timerms=0' was also my first thought, but it generating nearly 3ms, not two. Why? You may ask. Because you have free working timer and it one moment before pass 1ms. So when you reset Timerms following periods should pass:
- till end of just started millisecond till Timerms will be 1 (it is nearly 1ms)
- till Timerms will be 2 (it is lest say 1 ms)
- till Timerms will be 3 (it is lest say 1 ms)

So it is 3 ms.

Grzegorz Chrzanowski

danjperron commented:
Yes you are correct it is 3ms.  The minimum specification is 1ms and the maximum is 10ms. So 3ms is acceptable!

Commit:
Corrected length of servo PWN signal. In oryginal version period of signal was different and corelated with active PWM state. Now it is constant 20ms (defined in constant in RCServo.h).
Additionaly oryginal implementation of RC Servo PWM changes pin not configured as RC servo, when one of the pins was configured as RC-servo. It was problem for example when one of the pins was RC-servo and other was DHT-22. Now it is corrected.

danjperron commented:
Hi Grzegorz ,

   Just  wonder  how do you use my Pic multi purpose ?  Looks like you improve on it every times.

  This small pic version was the start of my 10 bit IO version which I made a small PCB of it.

  I tried to figure out how to accept your change but it get in conflict with your DHT22.c 
 modification . Anyway is in in pr3.

 thanks.

noisyfly commented:
Hi
I have seen that you have bigger version of the project. But I never look on that version more carefully. Sorry.

Your project for me is test platform for two other projects. One is more home automation. But for final project i will slightly change the firmware to make it more autonomous (light switches). There are three functions:
* light switches (digital In and digital Out), 
* thermostat sensor and heater control (DHT In and digital Out)
* and I think about humidity control in two versions (DHT In, digital or servo Out)
all this controlled from Home Assistant.

Second project it is implementation of pelco-d protocol for controlling camera position using Servos. For that I will rewrite firmware. But I will use already designed PCB. Maybe someone made this i will find out later. But I use your code to test mechanical platform for this.

Summarize:
* Your project look like you spend good time on this, and I feel that it will be good base for me (I don't have much time for development this days). I still think that it is very good.
* I have made the PCB for project (I can share it, if you want, but after tests, I made second version without prototype 
![win_20180225_12_47_40_pro](https://user-images.githubusercontent.com/33187000/36641065-59e51910-1a2a-11e8-8f4d-287152661f66.jpg)
).
* I changed your configuration tool to test servos (but not send it to you, but I will if you want to). 

About changes:
- this is my first time I provide more changes for someone project using github. Sorry I have no experience in that.
- I don't know what it is "pr3"
- could you explain me type of conflict with DHT22.c

Best regards

noisyfly commented:
I know "pr3" is the branch. I see it now. Sorry.

noisyfly commented:
If conflict is at void DHT22CycleIdle(void) near:

    **_Timerms=0;_**
     CurrentIOCycle= IO_CYCLE_START;

Simply remove **_Timerms=0;_** in your code.
